ports/raspberrypi/common-hal/displayio/ParallelBus.c

```diff
@@ -32,37 +32,128 @@
 #include "py/runtime.h"
 #include "shared-bindings/digitalio/DigitalInOut.h"
 #include "shared-bindings/microcontroller/__init__.h"
+#include "bindings/rp2pio/StateMachine.h"
+#include "common-hal/rp2pio/StateMachine.h"
+
+static const uint16_t parallel_program[] = {
+// .side_set 1
+// .wrap_target
+    0x6008, // out pins, 8 side 0
+    0xB042  // nop         side 1
+// .wrap
+};
 
 void common_hal_displayio_parallelbus_construct(displayio_parallelbus_obj_t* self,
     const mcu_pin_obj_t* data0, const mcu_pin_obj_t* command, const mcu_pin_obj_t* chip_select,
-    const mcu_pin_obj_t* write, const mcu_pin_obj_t* read, const mcu_pin_obj_t* reset) {
+    const mcu_pin_obj_t* write, const mcu_pin_obj_t* read, const mcu_pin_obj_t* reset, uint32_t frequency) {
+
+    // If we did not set frequency guess at 60Mhz
+    if (frequency == 0)
+        frequency = 60000000;
+
+    uint8_t data_pin = data0->number;
```

Comment:
I missed any verification of the pins being in order, as required for pio.  In #4329 (pending) we will add a convenience function for checking this.

Reply:
In the ParallelBus python interface you only give it data0 and the other pins are by definition of the interface immediately following.

Reply:
I missed that fact, thanks.